tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -130,42 +142,178 @@ private static ArgumentParser argParser() {
+            double requestSizeAvg = metrics.get(new MetricName("request-size-avg", "producer-metrics", "", tags)).value();
+            double recordQueueTimeAvg = metrics.get(new MetricName("record-queue-time-avg", "producer-metrics", "", tags)).value();
+            double requestLatencyAvg = metrics.get(new MetricName("request-latency-avg", "producer-metrics", "", tags)).value();
+            double recordsPerRequestAvg = metrics.get(new MetricName("records-per-request-avg", "producer-metrics", "", tags)).value();
+            double compressionRateAvg = metrics.get(new MetricName("compression-rate-avg", "producer-metrics", "", tags)).value();
+
+            System.out.println(String.format("Select_Rate_Avg:          %.2f", selectRateAvg));
+            System.out.println(String.format("Request_Rate_Avg:         %.2f", requestRateAvg));
+            System.out.println(String.format("Request_Latency_Avg:      %.2f", requestLatencyAvg));
+            System.out.println(String.format("Request_Size_Avg:         %.2f", requestSizeAvg));
+            System.out.println(String.format("Batch_Size_Avg:           %.2f", batchSizeAvg));
+            System.out.println(String.format("Records_Per_Request_Avg:  %.2f", recordsPerRequestAvg));
+            System.out.println(String.format("Record_Queue_Time_Avg:    %.2f", recordQueueTimeAvg));
+            System.out.println(String.format("Compression_Rate_Avg:     %.2f", compressionRateAvg));
+        }
+
+        private double avg(List<Double> samples) {
+            double sum = 0.0;
+            for (Double value : samples)
+                sum += value;
+            return samples.size() == 0 ? -1.0 : sum / samples.size();
+        }
+    }
+
     private static class Stats {
+        private ThreadStats[] threadStatsArr;
+        private long start;
+
+        public Stats(long numRecords, int reportingInterval, int numThreads) {
+            int numRecordsPerThread = (int) numRecords / numThreads;
+            int sampling = (int) (numRecordsPerThread / Math.min(numRecordsPerThread, 500000 / numThreads));
+            threadStatsArr = new ThreadStats[numThreads];
+            for (int i = 0; i < numThreads; i++) {
+                // If number of records cannot be exactly divided by num threads, some threads need to send one more
+                // record.
+                int numRecordsAdjustment = i < numRecords % numThreads ? 1 : 0;
+                threadStatsArr[i] = new ThreadStats(i, numRecordsPerThread + numRecordsAdjustment, reportingInterval, sampling);
+            }
+            this.start = System.currentTimeMillis();
+        }
+
+        public ThreadStats threadStats(int i) {
+            return threadStatsArr[i];
+        }
+
+        public void printTotal() {
+            long elapsed = System.currentTimeMillis() - start;
+            long totalCount = 0L;
+            long totalBytes = 0L;
+            long maxLatency = 0L;
```

Comment:
It is probably good to always have ms in the name variable name that represents time in ms. Probably not a big deal here since this is local variable.